kakarot_scripts/utils/kakarot.py

```diff
@@ -399,21 +406,26 @@ def _get_matching_logs_for_event(event_abi, log_receipts) -> List[dict]:
     return logs
 
 
-def _wrap_kakarot(fun: str, caller_eoa: Optional[Account] = None):
+def _wrap_kakarot(fun: Optional[str] = None, caller_eoa: Optional[Account] = None):
     """Wrap a contract function call with the Kakarot contract."""
 
     async def _wrapper(self, *args, **kwargs):
-        abi = self.get_function_by_signature(fun).abi
         gas_price = kwargs.pop("gas_price", DEFAULT_GAS_PRICE)
         gas_limit = kwargs.pop("gas_limit", TRANSACTION_GAS_LIMIT)
         value = kwargs.pop("value", 0)
         caller_eoa_ = kwargs.pop("caller_eoa", caller_eoa)
         max_fee = kwargs.pop("max_fee", None)
-        calldata = self.get_function_by_signature(fun)(
-            *args, **kwargs
-        )._encode_transaction_data()
 
-        if abi["stateMutability"] in ["pure", "view"]:
+        if fun is not None:
```

Comment:
enables sending txs to contracts that are not function invokations, through `contract.w3.eth.send_transaction`